### PR TITLE
Task/allow holder select disclosures

### PR DESCRIPTION
**Pull request by vijayshiyani**
When creating a keybinding SD JWT to preset to verifier, the holder can choose which disclosers to reveal.

### CODE REVIEW

File: src/holder.ts [hunk 1]
```diff
@@ -28,8 +36,13 @@ export class Holder {
    * @throws An error if the key binding JWT cannot be created.
    * @returns The key binding JWT.
    */
-  async getKeyBindingJWT(aud: string, nonce: string = generateNonce()): Promise<{ keyBindingJWT: JWT; nonce: string }> {
+  async getKeyBindingJWT(
+    aud: string,
+    nonceGenerator: NonceGenerator = nonceGeneratorCallbackFn(),
+  ): Promise<{ keyBindingJWT: JWT; nonce: string }> {
     try {
+      const nonce = nonceGenerator();
```
Comment (ragnika) [inline src/holder.ts:44]: @vijayshiyani my interpretations might be wrong, but I think the nonce is meant to be provided by the verifier
https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-00.html#section-6.1-2.1.1

So the holder would have to include it as is in the kb jwt

Comment (vijayshiyani) [reply src/holder.ts:44]: @ragnika correct, will fix it. thanks for identifying it.

File: src/holder.ts [hunk 2]
```diff
@@ -60,6 +73,8 @@ export class Holder {
     aud: string,
```
Comment (linasi) [inline src/holder.ts:73]: I would think more about API for this function. Now 5 inline params feel very confusing.

**Review by linasi — APPROVED** [review]
👍

File: src/holder.ts (hunk 2 above)
Comment (vijayshiyani) [reply src/holder.ts:73]: true, nonce, and audience should be an optional input. 
`  async presentVerifiableCredentialSDJWT(
    sdJWT: JWT,
    keyBindingJWTVerifier: KeyBindingVerifier,
    disclosedList: DisclosedList[],
    options?: { nonce?: string; audience?: string },
  ):`

Comment (vijayshiyani) [reply src/holder.ts:73]: I concluded the following would be a better signature, 
KeyBindingVerification can be optional. 

`/**
   * Presents a VC SD-JWT with a key binding JWT.
   * @param sdJWT The SD-JWT to present.
   * @param disclosedList The list of disclosed claims.
   * @param options The options to use.
   * @param options.nonce The nonce to use.
   * @param options.audience The verifier to present the VC SD-JWT to. e.g. https://example.com/verifier
   * @param options.keyBindingVerifierCallbackFn The callback function to verify the key binding JWT with the holder public key.
   * @throws An error if the VC SD-JWT cannot be presented.
   * @returns The VC SD-JWT with the key binding JWT.
   */
  async presentVerifiableCredentialSDJWT(
    sdJWT: JWT,
    disclosedList: DisclosedList[],
    options?: { nonce?: string; audience?: string; keyBindingVerifierCallbackFn?: KeyBindingVerifier },
  ): Promise<{ vcSDJWTWithkeyBindingJWT: JWT; nonce?: string }> {`

**Pull request merged**